examples/ecs/nondeterministic_system_order.rs

```diff
@@ -0,0 +1,85 @@
+//! These observable conflicts are called **system execution order ambiguities**.
+//!
+//! This example demonstrates how you might detect and resolve (or silence) these ambiguities.
+
+use bevy::{app::AppExit, ecs::schedule::ReportExecutionOrderAmbiguities, prelude::*};
+
+fn main() {
+    App::new()
+        // This resource controls the reporting strategy for system execution order ambiguities
+        .insert_resource(ReportExecutionOrderAmbiguities)
+        .init_resource::<A>()
+        .init_resource::<B>()
+        // This pair of systems has an ambiguous order,
+        // as their data access conflicts, and there's no order between them.
+        .add_system(reads_a)
+        .add_system(writes_a)
+        // This trio of systems has conflicting data access,
+        // but it's resolved with an explicit ordering
+        .add_system(adds_one_to_b)
+        .add_system(doubles_b.after(adds_one_to_b))
+        // This system isn't ambiguous with adds_one_to_b,
+        // due to the transitive ordering created.
+        .add_system(reads_b.after(doubles_b))
+        // This system will conflict with all of our writing systems
+        // but we've silenced its ambiguity with adds_one_to_b.
+        // This should only be done in the case of clear false positives:
+        // leave a comment in your code justifying the decision!
+        .add_system(reads_a_and_b.ambiguous_with(adds_one_to_b))
+        // Be mindful, internal ambiguities are reported too!
+        // If there are any ambiguities due solely to DefaultPlugins,
+        // or between DefaultPlugins and any of your third party plugins,
+        // please file a bug with the repo responsible!
+        // Only *you* can prevent nondeterministic bugs due to greedy parallelism.
+        .add_plugins(DefaultPlugins)
+        .run();
+}
+
+#[derive(Resource, Debug, Default)]
+struct A(usize);
+
+#[derive(Resource, Debug, Default)]
+struct B(usize);
+
+// Data access is determined solely on the basis of the types of the system's parameters
+// Every implementation of the `SystemParam` and `WorldQuery` traits must declare which data is used
+// and whether or not it is mutably accessed.
+fn reads_a(_a: Res<A>) {}
+
+fn writes_a(mut a: ResMut<A>) {
+    a.0 += 1;
+}
+
+fn adds_one_to_b(mut b: ResMut<B>) {
+    b.0 = b.0.saturating_add(1);
+}
+
+fn doubles_b(mut b: ResMut<B>) {
+    // This will overflow pretty rapidly otherwise
+    b.0 = b.0.saturating_mul(2);
+}
+
+fn reads_b(b: Res<B>) {
+    // This invariant is always true,
+    // because we've fixed the system order so doubling always occurs after adding.
+    assert!((b.0 % 2 == 0) || (b.0 == usize::MAX));
+}
+
+fn reads_a_and_b(a: Res<A>, b: Res<B>, mut app_exit: EventWriter<AppExit>) {
+    dbg!(a.0, b.0);
+
+    // Break early to make sure the debug output is easily seen
+    if b.0 > 10 {
+        app_exit.send_default();
+    }
```

Comment:
I don't think this example should be responsible of exiting

Reply:
If the example continues indefinitely, the user's console is rapidly spammed, and the ambiguities and debug output can't be seen.

I could add a timer, but I think that adds a ton of complexity.

Reply:
I could hide the dbg statement behind the if loop instead? But that seems kind of silly.